python/src/scipp/compat/dict.py

```diff
@@ -133,6 +133,7 @@ def _dict_to_variable(d):
     """
     Convert a python dict to a scipp Variable.
     """
+    _check_dict_input(d)
```

Comment:
Isn't `d = dict(d)` more pythonic?

Reply:
could be, but then you can get some strange error messages, e.g.
```Py
l = [1, 2, 3]
dict(l)

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-4e7674b4ef33> in <module>
      1 l = [1, 2, 3]
----> 2 dict(l)

TypeError: cannot convert dictionary update sequence element #0 to a sequence
```
But i guess it makes sense to see if something can be converted to a dict?

Reply:
I'd say so?

Reply:
Are you making a copy of the data when doing this? In the case where `d` contains a very large numpy array, could this become a problem?

Reply:
I don't believe `dict` makes deep copies, no.